NGitLab.Tests/Docker/GitLabDockerContainer.cs

```diff
@@ -316,14 +316,27 @@ async Task GenerateAdminToken(GitLabCredential credentials)
             // Create a token
             if (url == "/")
             {
+                ILocator formLocator;
+
                 TestContext.Progress.WriteLine("Creating root token");
 
-                await page.GotoAsync(GitLabUrl + "/-/profile/personal_access_tokens");
+                var tokenName = "GitLabClientTest-" + DateTime.UtcNow.ToString("yyyyMMdd-HHmmss", CultureInfo.InvariantCulture);
 
-                var formLocator = page.Locator("main#content-body form");
+                await page.GotoAsync(GitLabUrl + "/-/profile/personal_access_tokens");
 
-                var tokenName = "GitLabClientTest-" + DateTime.UtcNow.ToString("yyyyMMdd-HHmmss", CultureInfo.InvariantCulture);
-                await formLocator.GetByLabel("Token name").FillAsync(tokenName);
+                try
+                {
+                    await page.Locator("main[id='content-body'] button[data-testid='add-new-token-button']").ClickAsync(new LocatorClickOptions { Timeout = 5_000 });
+                    formLocator = page.Locator("main[id='content-body'] form[id='js-new-access-token-form']");
+                    await formLocator.Locator("input[data-testid='access-token-name-field']").FillAsync(tokenName);
+                }
+                catch
```

Comment:
Could the exception raised be specified for clarity (and to avoid catching others)? 

Is the GotoAsync in the catch necessary too? As you would already be on the correct page.

Reply:
You're completely right. I just had taken @louis-z's proposal. In the new implementation I'm just checking the version number, which we're running against ...